myBlog/tests/test_views.py

```diff
@@ -0,0 +1,72 @@
+from django.test import Client, TestCase
+from django.urls import reverse
+import uuid
+import json
+from myBlog.models import Author,Post,Comment,Friend
+from django.contrib.auth.models import User
+
+
+class TestViews(TestCase):
+    def setUp(self):
+        # https://stackoverflow.com/questions/2619102/djangos-self-client-login-does-not-work-in-unit-tests
+        self.user = User.objects.create(username='testuser')
+        self.user.set_password('test')
+
+        self.user.save()
+        self.client = Client()
+        self.client.login(username='testuser', password='test')
+
+        self.new_post_url = reverse('new_post')
+
+    # def test_New_Post_Handler_POST_API(self):
+    #     author = Author.objects.create(user=self.user,displayName="Test_author",github='https://github.com/terrence85561')
+    #
+    #     response = self.client.post(self.new_post_url,{
+    #         'title': 'POST1',
+    #         'description': 'post for testing',
+    #         'contentType': 'text/plain',
+    #         'visibility': 'PUBLIC',
+    #         'content': 'test'
+    #     })
+    #     self.assertEquals(response.status_code, 200)
+
+    def test_Post_Handler_GET_API(self):
+        # test get other author's post
+        # create an Author and User for this author
+        other_author_user_obj = User.objects.create(username='other_author_user_obj')
+        other_author_user_obj.set_password('test')
+        other_author_user_obj.save()
+        other_author_user_obj_client = Client()
+        other_author_user_obj_client.login(username='other_author_user_obj',password='test')
+        post_author = Author.objects.create(user=other_author_user_obj,displayName='author',github='https://github.com/terrence85561')
+
+        post1_postid=uuid.uuid4()
```

Comment:
when you create a new post, as a client, you don't know the id of the new post, it will be backend's job to generate an id for it